theano/tensor/sort.py

```diff
@@ -62,39 +62,31 @@ def grad(self, inputs, output_grads):
             self, 0, axis,
             "Currently, we only implement the gradient on sort for vector"
             " matrix (and axis is None or 0) and tensor3")
```

Comment:
Now, I think you implemented all supported cases. So I would get rid of that default "not implemented" message, and maybe put it in an "else" block below, changing the message to say that this is not a supported case.